src/target/drivers/filesystems/syscalls.c

```diff
@@ -33,49 +33,47 @@ extern void printstr(char *, int);
 #define dbgMsec()          ((DEBUG_SYSCALLS) ? CLOCK_getms() : 0)
 
 static DIR   dir;
+
 #ifdef MEDIA_DRIVE
-static FATFS fat[2];
+static DRIVE drive[2];
```

Comment:
I would prefer to not have another magic number here. maybe we can move the definition from petit_io.c to here. but I am fine you fist get this in.

Reply:
I want to get rid of the entire MEDIA special case, but it requires passing a file-pointer to the filesystem layer.  I'm not going to make that change now.